services/containerregistry/refreshtoken.go

```diff
@@ -0,0 +1,112 @@
+package containerregistry
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// ExchangeRefreshToken exchanges an Azure AD token for an Azure Container Registry refresh token
+func (c *ContainerRegistryClient) ExchangeRefreshToken(ctx context.Context) (string, *RefreshTokenClaims, error) {
```

Comment:
As per above comment on implementing auth.Authorizer, can we return `*oauth2.Token, error` here? Also, this might be better as `ObtainRefreshToken`?

Reply:
Maybe, I'm not sure yet. It's not implemented as a standard OAuth2 endpoint (at least to my understanding - may be wrong).

Naming is hard and I am bad at it. I welcome any and all name suggestions 😁

Reply:
As a continuation on the previous comment, I'll wait with this until you've taken a look at if you think `*oauth2.Token` is applicable here. When it comes to naming, `Exchange` is what they used in the Azure Go SDK for this. Naming is hard..